insights/parsers/messages.py

```diff
@@ -86,3 +86,23 @@ def get(self, s):
                     msg_info['procname'] = info_splits[4]
                 r.append(msg_info)
         return r
+
+    def get_after(self, timestamp, lines=None):
+        """
+        Get a list of lines after the given time stamp.
+
+        Parameters:
+            timestamp(datetime.datetime): log lines after this time are
+                returned.
+            lines(list): the list of log lines to search (e.g. from a get).
+                If not supplied, all available lines are searched.
+
+        Returns:
+            (list): The list of log lines with time stamps after the given
+                date and time.
+
+        *NOTE*: Because /var/log/messages timestamps by default have no year,
+        you must supply the year in your timestamp as 1900.  This may also
+        cause confusion around the end of the year.
```

Comment:
I assume that you do it this way because you can't assume the log is in chronological order?  If it was in order then you could just check the month value for a end-of-year rollover.